core/src/test/java/org/apache/iceberg/rest/TestRESTCatalog.java

```diff
@@ -72,4 +78,63 @@ protected boolean supportsNamespaceProperties() {
   protected boolean supportsServerSideRetry() {
     return true;
   }
+
+  /* RESTCatalog specific tests */
+
+  @Test
+  public void testConfigRoute() {
+    RESTClient testClient = new RESTClient() {
+      @Override
+      public void head(String path, Consumer<ErrorResponse> errorHandler) {
+        throw new UnsupportedOperationException("Should not be called for testConfigRoute");
+      }
+
+      @Override
+      public <T extends RESTResponse> T delete(String path, Class<T> responseType,
+                                               Consumer<ErrorResponse> errorHandler) {
+        throw new UnsupportedOperationException("Should not be called for testConfigRoute");
+      }
+
+      @Override
+      public <T extends RESTResponse> T get(String path, Class<T> responseType, Consumer<ErrorResponse> errorHandler) {
+        return (T) RESTCatalogConfigResponse
+            .builder()
+            .withDefaults(ImmutableMap.of())
+            .withOverrides(ImmutableMap.of("cache-enabled", "false"))
```

Comment:
Right now, this test uses disjoint property sets, but I'd like to see it validate the override behavior. I realize that's tested in the config response, but at least a basic check would be good here.

Reply:
Ok. I wasn't sure how much testing to add here because there are tests over there, but I agree it would be a good idea to have a bit more extensive testing here.

Reply:
Added more testing for defaults and overrides being applied. Between that and the tests actually in the `RESTCatalogConfigResponse`, things should be adequately tested.